File: python/credit/credit.py
cardNum_str = 0
while (True):
    try:

        cardNum_str = input("Number: ")
        if cardNum_str.isdigit() :
            cardNum =int(cardNum_str)
            break
        else:
            continue
    except ValueError:
        continue


if len(cardNum_str) == 15 and cardNum_str[:2] in ["34", "37"]:
    brand = "AMEX"
elif len(cardNum_str) == 16 and int(cardNum_str[:2]) in range(51,56):
    brand = "MASTERCARD"
elif (len(cardNum_str) == 13 or len(cardNum_str) == 16) and cardNum_str[0] == "4":
    brand = "VISA"
else:
    brand = "INVALID"

total = 0
reverse_cardNum = cardNum_str[::-1]
for i in range(len(cardNum_str)):
    digit = int(reverse_cardNum[i])
    if i % 2 == 1:
        product = digit * 2
        total += (product // 10) + (product % 10)
    else:
        total += digit
if total % 10 == 0:
    print(brand)
else:
    print("INVALID")

File: C/old/credit.c
#include <stdio.h>
#include <cs50.h>

int main(void)
{
    long number= get_long("Number: ");
    long original = number;
    int sum = 0;
    int type= 0;
    long temp= number;
    long digit = 0;
    while (temp >= 100)
    {
        temp /= 10;
    }
// digit is 1st digit, temp is first 2
    digit = temp /10;


    int position = 0;
    while (number > 0)
    {
        int last_digit = number % 10;

        if (position % 2 == 1)
        {
            int product = last_digit * 2;
            sum += (product /10) + (product %10);
        }
        else
        {
            sum += last_digit;
        }

        number /= 10;
        position++;
    }
    for (; original >= 1; original/=10)
    {
        type++;
    }
 if (sum % 10 == 0)
    {

    if ((digit == 4) && (type == 13 || type == 16))
    {
        printf("VISA\n");
    }
    else if ((temp >= 51 && temp <= 55) && type == 16)
    {
        printf("MASTERCARD\n");
    }

    else if ((temp == 31 || temp == 37) && type == 15)
    {
        printf("AMEX\n");
    }
    else
    {
        printf("INVALID\n");
    }
}
else{
    printf("INVALID\n");}

}

